samples/cpp/benchmark_app/main.cpp

```diff
@@ -1414,6 +1414,9 @@ int main(int argc, char* argv[]) {
             }
         }
 
+        // release static oneDNN cache used for inference
+        compiledModel.release_memory();
```

Comment:
Why not in `~CompiledModel()`?

If it stays here, `release_memory()` should be called in case of exception. Fix with RAII. Python should be updated as well as all samples